### PR TITLE
fix(dsv): strip bom from csv/tsv files if necessary

### PR DESCRIPTION
**Pull request by CrazyOrr**
## Rollup Plugin Name: `dsv`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:

### Description

Excel saves UTF-8 CSV files with a BOM. When such files are parsed, the key corresponding to the first column has `0xFEFF` as its first character, which leads to `keyName` became `"keyName"` (double-quoted).

As [this reply](https://github.com/d3/d3-dsv/issues/27#issuecomment-302151457) suggest, we should use [strip-bom](https://github.com/sindresorhus/strip-bom) to remove the BOM if necessary.

### CODE REVIEW

**Review by shellscape — APPROVED** [review]
@CrazyOrr looks like there's a conflict with pnpm-lock.yaml after a recent merge. please run `pnpm i` in the repo root directory and we'll be good to merge pending tests running in CI.

**Comment by CrazyOrr** [comment]
Done.

**Pull request merged**
